### PR TITLE
Refactor of GKE module to use upstream registry module

### PR DESCRIPTION
**Pull request by limed**
This PR replaces existing GKE module with the upstream terraform registry module, this also have additional niceties such as installing packages via helm.

### CODE REVIEW

File: gcp/gke/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+  ]
+}
+
+module "gke" {
+  source       = "github.com/mozilla-it/terraform-modules//gcp/gke?ref=master"
+  costcenter   = "1410"
+  environment  = "dev"
+  project_id   = local.project_id
+  cluster_name = local.cluster_name
```
Comment (mozafrank) [inline gcp/gke/README.md:36]: "name" would be more standards friendly than "cluster_name".

File: gcp/gke/variables.tf [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
 
-variable "http_load_balancing_disabled" { default = false }
+variable "ip_range_services" {
+  default = ""
+}
 
-variable "istio_disabled" { default = true }
+variable "regional" {
+  description = "Configure cluster as a regional or zonal cluster"
+  default     = true
```
Comment (mozafrank) [inline gcp/gke/variables.tf:34]: do we really want regional by default? not opposed but we hadn't discussed this, and we're currently generally using zonal in gke, and regional costs more.

File: gcp/gke/README.md (hunk 1 above)
Comment (limed) [reply gcp/gke/README.md]: Fixed in 7b1b3f5c949de5f14cb306d2e68b26350b63ff8e

File: gcp/gke/variables.tf (hunk 2 above)
Comment (limed) [reply gcp/gke/variables.tf]: Fixed in 7b1b3f5c949de5f14cb306d2e68b26350b63ff8e

**Pull request merged**